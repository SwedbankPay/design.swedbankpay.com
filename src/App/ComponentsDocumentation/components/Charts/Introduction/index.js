import React from "react";

import { ComponentPreview } from "@docutils";
import CodeTags from "@components/CodeTags";

const BASENAME = process.env.basename;
const brand = process.env.brand;
const scriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg.js`;
const dashboardScriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg-dashboard.js`;
const styleUrl = `https://design.${brand}.com${BASENAME}styles/dg-style.css`;
const dashboardStyleUrl = `https://design.${brand}.com${BASENAME}styles/dg-dashboard.css`;

const QuickStart = () => (
    <>
        <h3 id="quick-start">Quick start</h3>
        <p>To quickly add the dashboard theme to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h2 id="css">CSS</h2>
        <p>Copy-paste the stylesheet <CodeTags type="primary" code={"<link>"} /> into your <CodeTags type="primary" code={"<head>"} /> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={styleUrl} />
            <link rel="stylesheet" href={dashboardStyleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h2 id="javascript">JavaScript</h2>
        <p>The dashboard theme requires a separate script to run properly. So add the dashboard script below the <CodeTags type="secondary" code="scripts" /> at the bottom
        of your <CodeTags type="primary" code={"<body>"} /> tag like this:</p>
        <ComponentPreview language="html" codeFigure>
            <script src={scriptUrl}></script>
            <script src={dashboardScriptUrl}></script>
        </ComponentPreview>
    </>
);

const Introduction = () => (
    <>
        <h2 id="introduction">Introduction</h2>
        <p className="lead">
            We are using <a href="https://www.chartjs.org/" target="_blank" rel="noopener noreferrer">chart.js</a> to generate the charts.
            Our script is mainly just a wrapper around that library to add the correct styling etc.
        </p>
        <p>Most options and settings available in the chart.js documentation will be available.</p>
        <p>For now, the following chart types are available:</p>
        <ul>
            <li><CodeTags type="secondary" code={"\"line\""} /></li>
            <li><CodeTags type="secondary" code={"\"bar\"/\"horizontalBar\""} /></li>
            <li><CodeTags type="secondary" code={"\"pie\"/\"doughnut\""} /></li>
        </ul>
        <QuickStart />
        {/* <Css /> */}
        <JavaScript />
    </>
);

export default Introduction;

/* For testing */
export { QuickStart, Css, JavaScript };
