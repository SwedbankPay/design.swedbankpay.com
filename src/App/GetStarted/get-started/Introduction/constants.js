export const designGuideUsers = [
	{
		name: "Checkout",
		icon: "close",
		text: "No, but are following the Design Guide",
	},
	{
		name: "Swedbankpay.se",
		icon: "close",
		text: "No",
	},
	{
		name: "Consumer Portal",
		icon: "check",
		text: "Yes",
	},
	{
		name: "Developer Portal",
		icon: "check",
		text: "Yes",
	},
	{
		name: "Merchant Portal",
		icon: "check",
		text: "Yes",
	},
	{
		name: "Card terminals",
		icon: "close",
		text: "No, but are following the Design Guide",
	},
	{
		name: "Onboarding Portals",
		icon: "check",
		text: "Yes",
	},
	{
		name: "Terminal Management Systems",
		icon: "check",
		text: "Yes",
	},
];
