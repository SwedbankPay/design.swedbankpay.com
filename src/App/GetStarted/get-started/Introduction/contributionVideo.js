import React from "react";

const ContributionVideo = () => (
    <div className="video"><iframe frameBorder="0" allowFullScreen className="responsive-iframe" src={"https://player.vimeo.com/video/727671542?h=b9aa778573&amp;badge=0&amp;autopause=0&amp;player_id=0&amp;app_id=58479"}/></div>
);

export default ContributionVideo;
