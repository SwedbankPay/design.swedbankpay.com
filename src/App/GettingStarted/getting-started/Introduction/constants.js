export const designGuideUsers = [
    {
        name: "Checkout",
        icon: "close",
        text: "No, but are following the Design Guide"
    },
    {
        name: "Swedbankpay.se",
        icon: "close",
        text: "No"
    },
    {
        name: "Consumer Portal",
        icon: "check",
        text: "Yes"
    },
    {
        name: "Developer Portal",
        icon: "check",
        text: "Yes"
    },
    {
        name: "Card terminals",
        icon: "close",
        text: "No, but are following the Design Guide"
    },
    {
        name: "Ecom Merchant Admin",
        icon: "check",
        text: "Yes"
    },
    {
        name: "Ecom Owner Admin",
        icon: "check",
        text: "Yes"
    },
    {
        name: "Onboarding Signup",
        icon: "check",
        text: "Yes, is currently implemented the Design Guide"
    }
];
