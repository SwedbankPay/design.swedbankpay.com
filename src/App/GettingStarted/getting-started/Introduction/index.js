import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const BASENAME = process.env.basename;
const brand = process.env.brand;
const brandTitle = process.env.brandTitle;
const scriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg.js`;
const styleUrl = `https://design.${brand}.com${BASENAME}styles/dg-style.css`;

const QuickStart = () => (
    <>
        <h2 id="quick-start">Quick start</h2>
        <p>To quickly add the {brandTitle} DesignGuide to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h2 id="css">CSS</h2>
        <p>Copy-paste the stylesheet <CodeTags type="primary" code={"<link>"} /> into your <CodeTags type="primary" code={"<head>"} /> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={styleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h2 id="javascript">JavaScript</h2>
        <p>Many of our components require the use of JavaScript to function. Place the following <CodeTags type="primary" code={"<script>"} /> tag near the end of your pages, right before the
        closing <CodeTags type="primary" code={"</body>"} /> tag, to enable them.</p>
        <ComponentPreview language="html" codeFigure>
            <script src={scriptUrl}></script>
        </ComponentPreview>
    </>
);

const Introduction = () => (
    <DocContainer docToc>
        <p className="lead">Getting started with the {brandTitle} DesignGuide.</p>
        <QuickStart />
        <Css />
        <JavaScript />
    </DocContainer>
);

export default Introduction;

/* For testing */
export { QuickStart, Css, JavaScript };
