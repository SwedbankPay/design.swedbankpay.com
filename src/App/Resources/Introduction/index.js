import React from "react";

import { DocContainer } from "@docutils";

const brandTitle = process.env.brandTitle;

const Introduction = () => (
    <DocContainer docToc>
        <p className="lead">Getting started with the {brandTitle} Resources. Resources and guidelines on how to use the provided resources can be found here.</p>
    </DocContainer>
);

export default Introduction;
