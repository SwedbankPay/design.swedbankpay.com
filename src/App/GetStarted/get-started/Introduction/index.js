import React from "react";
import { Link } from "react-router-dom";
import { designGuideUsers } from "./constants";
import Lightbox from "@components/Lightbox";

import { DocContainer } from "@docutils";
import ContributionVideo from "./contributionVideo";

const basename = process.env.basename;
const brandTitle = process.env.brandTitle;

const GetStarted = () => (
    <section>
        <p>Below we have compiled a list of things you need to have and know in order to get started with our design guide.</p>

        <div className="component-overview">
            <Link to="/get-started/for-developers" className="cards cards-primary">
                <div className="cards-icon">
                    <i className="material-icons">code</i>
                </div>
                <div className="cards-content">
                    <span className="h4">For developers</span><span>Learn more about how to setup and use our system.</span>
                </div>
                <i className="material-icons">arrow_forward</i>
            </Link>
            <Link to="/get-started/for-designers" className="cards cards-primary">
                <div className="cards-icon">
                    <i className="material-icons">brush</i>
                </div>
                <div className="cards-content">
                    <span className="h4">For designers</span><span>Get ready to use the Design Guide when designing in {brandTitle}.</span>
                </div>
                <i className="material-icons">arrow_forward</i>
            </Link>
        </div>
    </section>
);

const WhatIsDG = () => (
    <section id="what-design-guide">
        <h2 id="what-is-dg">What is the Design Guide?</h2>
        <p>The {brandTitle} design system consists of two parts; the Figma designs and the web Design Guide (where you are right now). The Design Guide is the ready-to-use part of the design system and the Figma design library is for designing and prototyping.</p>
        <p>The Design Guide consists of more than design. It contains all the parts of the design system that actually can be used and implemented by developers working with {brandTitle} products.</p>

        <section>
            <h2>Working with the Design Guide</h2>
            <p>The Design Guide is an organically growing system that evolves with the products. Product teams can use the Design Guide for brand identity, components, guidelines, and documentation regarding accessibility, copywriting, and more. Product teams also contribute to the system by suggesting changes and improvements to already existing components and guidelines, as well as by adding completely new ones.</p>
            <Lightbox imgSrc={`${basename}img/documentation/introduction/working-dg.png`} className="img-fluid" alt="Illustation of how product teams can use the design system and how to contribute."/>
        </section>
        <section>
            <h3>Contributing to the system</h3>
            <p>Components and assets that can be reused in other products should be added to the system for others to use. However, some components that fulfill an isolated need for a specific product will remain local for that product only. This informational video explains the design workflow and snowflake process, and how your product team can contribute to the Design Guide:</p>
            <ContributionVideo/>
            <a className="icon-link mt-3" target="_blank" rel="noopener noreferrer" href={"http://www.figma.com/proto/p1QcsYCJ3T466As9xQQGBM/Design-Guide-Mantra?page-id=11%3A4123&node-id=233%3A4515&viewport=306%2C372%2C0.1&scaling=min-zoom&starting-point-node-id=233%3A4515"}><i className="material-icons mr-2" aria-hidden="true">launch</i>View the contribution process prototype</a>
        </section>
        <section>
            <h3>Products using the Design Guide</h3>
            <p>An interface may look right according to the Design Guide, our brand, and the design system in general, but that does not mean that it, per definition, “uses the Design Guide“. The definition of using our Design Guide is the technical implementation and using the components taken straight from the Design Guide.</p>
            <p>When an adjustment or change occurs with one of the Design Guide components, all the projects using said component will have the changes applied to their product automatically. This way, the Design Guide enables a cohesive and unified design across the organization, which saves time for everyone involved.</p>
            {brandTitle === "Swedbank Pay" && <section>
                <h3>Products using the Design Guide</h3>
                <p>Because there’s a difference of following the design guidelines and using the Design Guide, not all {brandTitle} products per definition uses the Design Guide. </p>
                <table className="table table-plain">
                    <thead>
                        <tr>
                            <th>Product</th>
                            <th>Using Design Guide</th>
                        </tr>
                    </thead>
                    <tbody>
                        {designGuideUsers.map(product => (
                            <tr key={product.name}>
                                <td>{product.name}</td>
                                <td><i className="material-icons mr-3">{product.icon}</i>{product.text}</td>
                            </tr>
                        ))}
                    </tbody>
                </table>
            </section>}
        </section>
    </section>
);

const Feedback = () => (
    <section>
        <h2 id="feedback">Feedback</h2>
        <p>We want to do our best to provide top-quality components for you to use, but this is only possible with your help. If you have any suggestions for how we could potentially improve a component, please don´t hesitate to reach out. If you have questions, feedback, or want to know what’s new in the Design Guide, feel free to join us in the #design-guide-general channel on Slack.</p>
    </section>
);

const Introduction = () => (
    <DocContainer>
        <p className="lead">We are so glad to see that you are interested in learning more about the {brandTitle} Design Guide. We know that it can be daunting to start with something new and complex. But don’t worry, we have got what you need!</p>
        <GetStarted />
        <WhatIsDG />
        <Feedback/>
    </DocContainer>
);

export default Introduction;

export { GetStarted, WhatIsDG, Feedback };
