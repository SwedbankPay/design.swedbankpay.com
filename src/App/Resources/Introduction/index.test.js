import React from "react";
import { shallow } from "enzyme";

import Introduction from "./index";

describe("Resources: Introduction", () => {
    it("is defined", () => {
        expect(Introduction).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Introduction />);

        expect(wrapper).toMatchSnapshot();
    });
});
