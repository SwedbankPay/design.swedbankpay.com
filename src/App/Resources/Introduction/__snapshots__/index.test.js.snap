// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources: Introduction renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Getting started with the 
    TESTBRAND
     Resources. Resources and guidelines on how to use the provided resources can be found here.
  </p>
</DocContainer>
`;
